src/proxy/http/router.rs

```diff
@@ -42,8 +42,9 @@ pub struct Stack<Req, Rec: Recognize<Req>, Mk> {
 
 pub struct Service<Req, Rec, Mk>
 where
-    Rec: Recognize<Req>,
-    Mk: rt::Make<Rec::Target>,
+    Rec: Recognize<Req> + Clone + Send,
+    <Rec as Recognize<Req>>::Target: Clone,
```

Comment:
so, it seems that all uses of Recognize requie that `Target: Clone`... You could probably reduce boilerplate by moving this contraint onto the trait definition?